Src/C#-SIMS/Inventory.cs

```diff
@@ -0,0 +1,217 @@
+            Console.Write("Enter the product name: ");
+            userName = Console.ReadLine();
+            Console.Write("Enter the product price: ");
+            parseSuccess = int.TryParse(Console.ReadLine(), out int productPrice);
+            price = parseSuccess ? productPrice : 0;
+            Console.Write("Enter the product quantity: ");
+            parseSuccess = int.TryParse(Console.ReadLine(), out int productQuantity);
+            quantity = parseSuccess ? productQuantity : 0;
+            products.Add(new Product(userName, price, quantity));
+            Console.Write(
+                "The product added successfully.....\nPress 1 to continue or 0 to exit to the main menu: "
+            );
+            userSelection = Console.ReadLine();
+        } while (userSelection != "0");
+        ShowMenu();
+    }
+
+    private static void ViewAllProducts()
+    {
+        initUI();
+        Console.WriteLine("****View all products****");
+
+        int index = 1;
+        foreach (var product in products)
+        {
+            Console.Write($"{index++}: ");
+            Console.WriteLine(product.ViewProductDetail());
+        }
+        ReturnToMainMenu();
+    }
+
+    private static void EditProduct()
+    {
+        initUI();
+        string? nameIO;
+        Console.WriteLine("****Edit a product****");
+        int index = 1;
+        foreach (var product in products)
+        {
+            Console.Write($"{index++}: ");
+            Console.WriteLine(product.Name);
+        }
+        Console.Write("PLease write the product's name: ");
+        nameIO = Console.ReadLine();
+        Product? productToEdit = products.Where(p => p.Name == nameIO).FirstOrDefault();
+        if (productToEdit != null)
+        {
+            Console.WriteLine("Please choose the number of what you want to change. (You can choose multiple choices, just separate them with commas.):");
+            Console.Write("1: Name\n2: Price\n3: Quantity\n: ");
+            string? optionsIO = Console.ReadLine();
+            List<string> itemsList = new List<string>(optionsIO?.Split(',') ?? Array.Empty<string>());
+            do
+            {
+                switch (itemsList[0])
+                {
+                    case "1":
+                        Console.Write("Please enter the new name: ");
+                        string? name = Console.ReadLine();
+                        productToEdit.Name = name;
+                        break;
+                    case "2":
+                        Console.Write("Please enter the new price: ");
+                        int price = int.Parse(Console.ReadLine() ?? "0");
+                        productToEdit.Price = price;
+                        break;
+                    case "3":
+                        Console.Write("Please enter the new quantity: ");
+                        int quantity = int.Parse(Console.ReadLine() ?? "0");
+                        productToEdit.Quantity = quantity;
```

Comment:
will updating these values be enough to update the product in the products list ?

Reply:
Hmm, why not? I mean, I first found the product reference, then I made my updates on that reference, which automatically changed it in the list too.